File: app/src/test/java/com/grace/placessearch/service/local/VenuesTest.java
package com.grace.placessearch.service.local;

import static org.assertj.core.api.Assertions.assertThat;

import com.grace.placessearch.common.data.model.SuggestedVenuesResponse;
import com.grace.placessearch.common.data.model.Venue;
import com.grace.placessearch.common.data.model.VenueResponse;
import com.grace.placessearch.common.data.model.VenuesResponse;

import org.junit.Test;

import java.util.List;

import retrofit2.adapter.rxjava.Result;
import rx.Observable;
import rx.observables.BlockingObservable;

public class VenuesTest extends PlacesTestBase {

    @Test
    public void testTrendingVenues() {
        Observable<Result<VenuesResponse>> observable = getPlaces().trendingVenues();
        BlockingObservable<Result<VenuesResponse>> bo = BlockingObservable.from(observable);
        VenuesResponse venuesResponse = bo.first().response().body();
        assertThat(venuesResponse).isNotNull();
        assertThat(venuesResponse.getVenueListResponse().getVenues().size()).isEqualTo(18);
    }

    @Test
    public void testSuggestedVenues() {
        Observable<Result<SuggestedVenuesResponse>> observable = getPlaces().searchForSuggestedVenues("cof");
        BlockingObservable<Result<SuggestedVenuesResponse>> bo = BlockingObservable.from(observable);
        SuggestedVenuesResponse venuesResponse = bo.first().response().body();
        assertThat(venuesResponse).isNotNull();
        assertThat(venuesResponse.getResponse().getVenues().size()).isEqualTo(25);

        // Test first venue
        List<Venue> suggestedVenues = venuesResponse.getResponse().getVenues();
        Venue venue = suggestedVenues.get(0);
        assertThat(venue).isNotNull();

        assertThat(venue.getId()).isEqualTo("52d456c811d24128cdd7bc8b");
        assertThat(venue.getName()).isEqualTo("Storyville Coffee Company");
        assertThat(venue.getLocation()).isNotNull();
        assertThat(venue.getCategories().size()).isEqualTo(2);
        assertThat(venue.getCategories().get(0).getId()).isEqualTo("4bf58dd8d48988d1e0931735");
        assertThat(venue.getCategories().get(0).getName()).isEqualTo("Coffee Shop");
        assertThat(venue.getCategories().get(1).getId()).isEqualTo("4d4b7105d754a06374d81259");
        assertThat(venue.getCategories().get(1).getName()).isEqualTo("Food");
    }

    @Test
    public void testGetVenue() {
        Observable<Result<VenueResponse>> observable = getPlaces().getVenue("cof");
        BlockingObservable<Result<VenueResponse>> bo = BlockingObservable.from(observable);
        VenueResponse venuesResponse = bo.first().response().body();
        assertThat(venuesResponse).isNotNull();

        // Test venue
        Venue venue = venuesResponse.getSingleVenueResponse().getVenue();
        assertThat(venue).isNotNull();

        assertThat(venue.getId()).isEqualTo("52d456c811d24128cdd7bc8b");
        assertThat(venue.getName()).isEqualTo("Storyville Coffee Company");
        assertThat(venue.getLocation()).isNotNull();
        assertThat(venue.getCategories().size()).isEqualTo(1);
        assertThat(venue.getCategories().get(0).getId()).isEqualTo("4bf58dd8d48988d1e0931735");
        assertThat(venue.getCategories().get(0).getName()).isEqualTo("Coffee Shop");
    }
}

File: app/src/test/kotlin/com/grace/placessearch/service/local/VenuesTest.kt
package com.grace.placessearch.service.local

import org.assertj.core.api.Assertions.assertThat

import org.junit.Test

import rx.observables.BlockingObservable

class VenuesTest : PlacesTestBase() {

    @Test
    fun testTrendingVenues() {
        val observable = places.trendingVenues()
        val bo = BlockingObservable.from(observable)
        val venuesResponse = bo.first().response().body()
        assertThat(venuesResponse).isNotNull()
        assertThat(venuesResponse.venueListResponse?.venues?.size).isEqualTo(18)
    }

    @Test
    fun testSuggestedVenues() {
        val observable = places.searchForSuggestedVenues("cof")
        val bo = BlockingObservable.from(observable)
        val venuesResponse = bo.first().response().body()
        assertThat(venuesResponse).isNotNull()
        assertThat(venuesResponse.response?.venues?.size).isEqualTo(25)

        // Test first venue
        val suggestedVenues = venuesResponse.response?.venues
        val venue = suggestedVenues!![0]
        assertThat(venue).isNotNull()

        assertThat(venue.id).isEqualTo("52d456c811d24128cdd7bc8b")
        assertThat(venue.name).isEqualTo("Storyville Coffee Company")
        assertThat(venue.location).isNotNull()
        assertThat(venue.categories.size).isEqualTo(2)
        assertThat(venue.categories[0].id).isEqualTo("4bf58dd8d48988d1e0931735")
        assertThat(venue.categories[0].name).isEqualTo("Coffee Shop")
        assertThat(venue.categories[1].id).isEqualTo("4d4b7105d754a06374d81259")
        assertThat(venue.categories[1].name).isEqualTo("Food")
    }

    @Test
    fun testGetVenue() {
        val observable = places.getVenue("cof")
        val bo = BlockingObservable.from(observable)
        val venuesResponse = bo.first().response().body()
        assertThat(venuesResponse).isNotNull()

        // Test venue
        val venue = venuesResponse.singleVenueResponse?.venue
        assertThat(venue).isNotNull()

        assertThat(venue?.id).isEqualTo("52d456c811d24128cdd7bc8b")
        assertThat(venue?.name).isEqualTo("Storyville Coffee Company")
        assertThat(venue?.location).isNotNull()
        assertThat(venue?.categories?.size).isEqualTo(1)
        assertThat(venue?.categories?.get(0)?.id).isEqualTo("4bf58dd8d48988d1e0931735")
        assertThat(venue?.categories?.get(0)?.name).isEqualTo("Coffee Shop")
    }
}
